### PR TITLE
Initial commit for app as per PAPP-25699

### PR DESCRIPTION
**Pull request by achandya-crest**
Integration pipeline: https://jenkins.engineering.phantom.us:8443/job/run-app-integration-tests/8285/console

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [x] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [x] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [x] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by jdemelo** [comment]
[Most of the playbook tests](https://cd.splunkdev.com/phantom-apps/app-tests/-/blob/PAPP-25699_initial_pytest_cloudconvert/suite/apps/cloudconvert/cloudconvert_sanity_test.py#L33) are marked with `xfail` because the cloudconvert [license limits us to 25 conversions per day](https://splunk.atlassian.net/browse/PINF-924). Is it a risk that we're preparing to release without running so many playbook tests?

**Comment by achandya-crest** [comment]
@jdemelo, We have just skipped all the playbook runs in the pytest pipeline because while releasing the app, the pytest runs 4times due to which it would definitely fail as the number of runs in all the playbooks exceed 25. We have already tested all the playbooks in the various rounds of app testing before releasing the app. Hence, we will make sure all playbooks are test cases are tested thoroughly before releasing the app. Also, once we have an account in future with higher conversion limit, we will be removing XFAIL from the pytest script.

File: cloudconvert.json [hunk 1]
```diff
@@ -0,0 +1,689 @@
+{
+    "appid": "5cf7f9aa-877a-45fb-8384-362d2b935db9",
+    "name": "Cloud Convert",
+    "description": "This app supports executing investigative and generic type of actions to convert the files to various formats",
```
Comment (ricardo-phantom) [inline cloudconvert.json:4]: which files?

Comment (achandya-crest) [reply cloudconvert.json]: Changed

**Comment by ricardo-phantom** [comment]
Integration tests are failing

File: cloudconvert.json [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+                    "primary": true,
+                    "contains": [
+                        "vault id"
+                    ],
+                    "order": 0
+                },
+                "filetype": {
+                    "description": "Type of the file in output",
+                    "data_type": "string",
+                    "value_list": [
```
Comment (alexa-phantom) [inline cloudconvert.json:69]: How is this list ordered? Might be useful to do alphabetical or most commonly used at top

File: cloudconvert_connector.py [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+            ret_val, response = self._make_rest_call(
+                url=url,
+                action_result=action_result,
+                method="get"
+            )
+            if phantom.is_fail(ret_val):
+                return action_result.get_status(), None
+            counter += sleep_seconds
+            for task in response.get("data"):
+                if task.get("job_id") == job_id:
```
Comment (alexa-phantom) [inline cloudconvert_connector.py:560]: Use single quotes for consistency

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: cloudconvert.json (hunk 2 above)
Comment (achandya-crest) [reply cloudconvert.json:69]: There are no such commonly used outputs formats. Hence, we used an API call https://api.cloudconvert.com/v2/convert/formats provided by cloud convert which returns the list of valid output formats and placed them in the value_list in the same order. It isn't mentioned anywhere whether this API returns most commonly used formats first or not. Also as value list provides valid suggestions for providing some input(For instance, On entering 'p' in dropdown parameter, all values in list starting with 'p' are returned), we haven't changed the list returned by the API.

File: cloudconvert_connector.py (hunk 3 above)
Comment (achandya-crest) [reply cloudconvert_connector.py]: Done

**Review by alexa-phantom — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**